jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java

```diff
@@ -117,6 +128,25 @@ public void onComplete(AsyncEvent event) throws IOException
             }   
         }
     };
+    
+    private final HttpChannel.Listener _shutdownListener = new HttpChannel.Listener()
+    {
+        @Override 
+        public void onResponseBegin(Request request)
+        {
+            if (_shutdown.isShutdown())
+            {
+                Connector connector = request.getHttpChannel().getConnector();
+                if (connector instanceof Graceful && ((Graceful)connector).isShutdown())
+                {
+                    MetaData.Response response = request.getHttpChannel().getCommittedMetaData();
+
+                    if (!response.getFields().contains(HttpHeader.CONNECTION,"close"))
```

Comment:
Use `HttpHeaderValue.CLOSE`?

Reply:
Can't we add this logic to the `HttpGenerator` instead here?
The `HttpGenerator` will ask whether the channel is shutting down, and will generate a `Connection: close`.